src/App.js

```diff
@@ -1,44 +1,58 @@
 import React from 'react';
-import Header from './Header';
-import Footer from './Footer';
+import Header from './Header.js';
+import Footer from './Footer.js';
 import 'bootstrap/dist/css/bootstrap.min.css';
 import {
   BrowserRouter as Router,
   Switch,
   Route
 } from "react-router-dom";
+import BestBooks from './BestBooks.js';
+import Profile from './Profile';
+import LoginButton from './LoginButton';
+import LoginForm from './LoginForm.js';
 
 class App extends React.Component {
 
   constructor(props) {
     super(props);
     this.state = {
       user: null,
+      email: '',
     }
   }
 
   loginHandler = (user) => {
+    user.preventDefault();
+
     this.setState({
-      user,
+      user: user.target[1].value,
+      email: user.target[0].value,
     })
   }
 
   logoutHandler = () => {
     this.setState({
       user: null,
-    })
+    });
   }
 
   render() {
     return (
       <>
+
         <Router>
           <Header user={this.state.user} onLogout={this.logoutHandler} />
           <Switch>
             <Route exact path="/">
-              {/* TODO: if the user is logged in, render the `BestBooks` component, if they are not, render the `Login` component */}
+              {this.state.user ? <BestBooks /> : <LoginButton onLogin={this.loginHandler} />}
```

Comment:
- line 48, we are rendering BestBooks if logged in, if not, we are shoing the button which has props of onLogin. Instead, can we pass the Login component here and not the button itself?